Mathlib/Dynamics/TopologicalEntropy/CoverEntropy.lean

```diff
@@ -0,0 +1,642 @@
+
+All is stated in the vocabulary of uniform spaces. For compact spaces, the uniform structure
+is canonical, so the topological entropy depends only on the topological structure. This will give
+a clean proof that the topological entropy is a topological invariant of the dynamics.
+
+A notable choice is that we define the topological entropy of a subset `F` of the whole space.
+Usually, one defines the entropy of an invariant subset `F` as the entropy of the restriction of the
+transformation to `F`. We avoid the latter definition as it would involve frequent manipulation of
+subtypes. Our version directly gives a meaning to the topological entropy of a subsystem, and a
+single theorem (`subset_restriction_entropy` in `TopologicalEntropy.Morphism`) will give the
+equivalence between both versions.
+
+Another choice is to give a meaning to the entropy of `∅` (it must be `-∞` to stay coherent) and to
+keep the possibility for the entropy to be infinite. Hence, the entropy takes values in the extended
+reals `[-∞, +∞]`. The consequence is that we use `ℕ∞`, `ℝ≥0∞` and `EReal` numbers.
+
+## Main definitions
+- `IsDynCoverOf`: property that dynamical balls centered on a subset `s` cover a subset `F`.
+- `coverMincard`: minimal cardinal of a dynamical cover. Takes values in `ℕ∞`.
+- `coverEntropyInfUni`/`coverEntropySupUni`: exponential growth of `coverMincard`. The former is
+defined with a `liminf`, the later with a `limsup`. Take values in `EReal`.
+- `coverEntropyInf`/`coverEntropySup`: supremum of `coverEntropyInfUni`/`coverEntropySupUni` over
+all entourages (or limit as the entourages go to the diagonal). These are Bowen-Dinaburg's
+versions of the topological entropy with covers. Take values in `EReal`.
+
+## Tags
+cover, entropy
+
+## TODO
+The most painful part of many manipulations involving topological entropy is going from
+`coverMincard` to `coverEntropyInfUni`/`coverEntropySupUni`. It involves a logarithm, a division, a
+`liminf`/`limsup`, and multiple coercions. The best thing to do would be to write a file on
+"exponential growth" to make a clean pathway from estimates on `coverMincard` to estimates on
+`coverEntropy`/`coverEntropy'`. It would also be useful in other similar contexts, including the
+definition of entropy using nets.
+
+Get versions of the topological entropy on (pseudo-e)metric spaces.
+-/
+
+namespace Dynamics
+
+open Set Uniformity UniformSpace
+
+variable {X : Type*}
+
+/-! ### Dynamical covers -/
+
+/-- Given a subset `F`, an entourage `U` and an integer `n`, a subset `s` is a `(U, n)`-
+dynamical cover of `F` if any orbit of length `n` in `F` is `U`-shadowed by an orbit of length `n`
+of a point in `s`.-/
+def IsDynCoverOf (T : X → X) (F : Set X) (U : Set (X × X)) (n : ℕ) (s : Set X) : Prop :=
+  F ⊆ ⋃ x ∈ s, ball x (dynEntourage T U n)
+
+lemma dynCover_antitone_time {T : X → X} {F : Set X} {U : Set (X × X)} {m n : ℕ} (m_n : m ≤ n)
```

Comment:
If your lemma statement does not mention `antitone`, your lemma name shouldn't either. Also, since it deals with `IsDynCoverOf`, it should mention `isDynCoverOf`. Since you have an assumption `IsDynCoverOf`, I would probably call it `IsDynCoverOf.of_le` to make sure it can be used through dot notation. 

Could you review all lemma names in the PR according to these conventions?

Reply:
@sgouezel: Agree with the general point. There's however another naming issue related to these lemmas. There are three variables you can play with: `F`, `U` and `n`.

Something like `IsDynCoverOf.of_le` can work for the variable `n`, but how can I differentiate between `F` and `U`? I can't call them both `IsDynCoverOf.of_subset`.

Reply:
BTW, done as well as I could. I've also simplified a few things (avoid using sInf, and removed a small lemma which wasn't used).